drivers/input/input_tsc_keys.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2024 Arif Balik <arifbalik@outlook.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT tsc_keys
+
+#include <autoconf.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/misc/stm32_tsc/stm32_tsc.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/spsc_lockfree.h>
+#include <zephyr/sys/spsc_pbuf.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/ring_buffer.h>
+
+LOG_MODULE_REGISTER(tsc_keys, CONFIG_INPUT_LOG_LEVEL);
+
+struct input_tsc_keys_child_data {
+	uint32_t buffer[CONFIG_INPUT_TSC_KEYS_BUFFER_WORD_SIZE];
+	struct ring_buf rb;
+	enum {
+		INPUT_TSC_KEYS_EXPECT_DOWN,
+		INPUT_TSC_KEYS_EXPECT_UP,
+	} state;
+};
+
+struct input_tsc_keys_child_config {
+	char *label;
+	uint8_t group_index;
+	int32_t noise_threshold;
+	int zephyr_code;
+	struct input_tsc_keys_child_data *data;
+};
+
+struct input_tsc_keys_data {
+	struct k_timer sampling_timer;
+};
+
+struct input_tsc_keys_config {
+	const struct device *tsc_dev;
+	stm32_tsc_callback_t tsc_callback;
+	uint32_t sampling_interval_ms;
+	struct input_tsc_keys_child_config *children;
+	uint8_t child_num;
```

Comment:
I'd use `child_count`, I know the devicetree API is called "num" but this sounds a bit confusing